language_model/rnn_train.py

```diff
@@ -0,0 +1,104 @@
+import sys
+import sqlite3
+import paddle.v2 as paddle
+import ptb
+import gzip
+
+
+def rnn_net(dict_dim, is_generating=False):
+    # dimension of word vector
+    word_vector_dim = 1024
+    #dimension of hidden unit in GRU Encoder network
+    encoder_size = 1024
+
+    # Encoder
+    src_word_id = paddle.layer.data(
+        name='source_word',
+        type=paddle.data_type.integer_value_sequence(dict_dim))
+    target_word_id = paddle.layer.data(
+        name='target_word',
+        type=paddle.data_type.integer_value_sequence(dict_dim))
+
+    src_embedding = paddle.layer.embedding(
+        input=src_word_id,
+        size=word_vector_dim,
+        param_attr=paddle.attr.ParamAttr(name='_source_word_embedding'))
+
+    src_forward = paddle.networks.simple_gru(
+        input=src_embedding, size=encoder_size)
+    src_backward = paddle.networks.simple_gru(
+        input=src_embedding, size=encoder_size, reverse=True)
+    encoded_vector = paddle.layer.concat(input=[src_forward, src_backward])
+    output = paddle.layer.fc(
+        input=encoded_vector,
+        size=dict_dim,
+        act=paddle.activation.Softmax(),
+        param_attr=paddle.attr.Param(initial_std=0.01))
+    # calc cost
+    cost = paddle.layer.classification_cost(input=output, label=target_word_id)
+
+    return cost
+
+
+def main():
+    paddle.init(use_gpu=False, trainer_count=8)
+    is_generating = False
+
+    # source and target dict dim.
+    word_dict = ptb.build_dict()
+    dict_size = len(word_dict)
+    # train the network
+    if not is_generating:
+        cost = rnn_net(dict_size)
+        parameters = paddle.parameters.create(cost)
+
+        # define optimize method and trainer
+        optimizer = paddle.optimizer.Adam(
+            learning_rate=5e-3,
+            regularization=paddle.optimizer.L2Regularization(rate=8e-4))
+
+        trainer = paddle.trainer.SGD(
+            cost=cost, parameters=parameters, update_equation=optimizer)
+
+        # define data reader
+        ptb_reader = paddle.batch(
+            paddle.reader.shuffle(ptb.seq_train(word_dict, 100), buf_size=8192),
```

Comment:
I wonder whether this script can run or not. I am confusing with the input data format and how to provide data to PaddlePaddle models from the raw input data. 

We cannot provide an example that cannot run or an example that we cannot explain how to run to the users. 

Is the `ptb`  module a dataset you want to add into paddle.dataset for `the Penn Treebank` dataset?  If so, it should be put into paddle.dataset. The name `ptb` is very confusing with the python  traceback module.